src/netex-convertor/netexGenerator.ts

```diff
@@ -117,49 +126,50 @@ const netexGenerator = (
 
     const updateResourceFrame = (resourceFrame: NetexObject): NetexObject => {
         const resourceFrameToUpdate = { ...resourceFrame };
-        const operatorPublicName = isBaseSchemeOperatorInfo(baseOperatorInfo)
-            ? baseOperatorInfo.schemeOperatorName
-            : baseOperatorInfo.operatorPublicName;
 
-        resourceFrameToUpdate.id = `epd:UK:${coreData.operatorIdentifier}:ResourceFrame_UK_PI_COMMON:${coreData.operatorIdentifier}:op`;
+        resourceFrameToUpdate.id = `epd:UK:${coreData.operatorIdentifier}:ResourceFrame_UK_PI_COMMON${`:${
+            !pointToPoint ? `${coreData.operatorIdentifier}:` : ''
+        }`}op`;
         resourceFrameToUpdate.codespaces.Codespace.XmlnsUrl.$t = coreData.website;
         resourceFrameToUpdate.dataSources.DataSource.Email.$t = baseOperatorInfo.ttrteEnq;
         resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[0].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.ref =
             coreData.nocCodeFormat;
-        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[0].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t = operatorPublicName;
+        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[0].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t =
+            coreData.operatorName;
         resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.ref =
             coreData.nocCodeFormat;
-        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t = operatorPublicName;
+        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t =
+            coreData.operatorName;
+
         resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.id = coreData.brandingId;
-        resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Name.$t = operatorPublicName;
+        resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Name.$t = coreData.operatorName;
         resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Url.$t = coreData.website;
 
-        if (
-            ticket.type === 'multiOperator' &&
-            (isMultiOperatorGeoZoneTicket(ticket) || isSchemeOperatorTicket(ticket))
-        ) {
+        if (isMultiOperator && (isMultiOperatorGeoZoneTicket(ticket) || isSchemeOperatorTicket(ticket))) {
             const nocs = [...ticket.additionalNocs];
-            if (isMultiOperatorGeoZoneTicket(ticket)) {
-                nocs.push(ticket.nocCode);
+            if (isGeoZoneTicket(ticket)) {
```

Comment:
Since the SchemeOperatorTicket type can also be a GeoZoneTicket, they could get into this if at which point the nocCode will be undefined since they don't exist on that type, we could do `if(!isSchemeOperatorTicket(ticket))` instead unless there are circumstances that we want scheme tickets to get into this if?